File: snapchat/profile/oc/CodiaDemoOC/src/profileProfile/ProfileGroup3.m
//
//  ProfileGroup3.m
//
//  Created by codia-figma
//

#import <UIKit/UIKit.h>
#import <Masonry/Masonry.h>
#import "ProfileGroup3.h"
#import <UIKit/UIImageView.h>

@interface ProfileGroup3 ()

/// UIImageView: find friend
@property (nonatomic, strong) UIImageView *profileImageFindFriend;
@property (nonatomic, strong) UILabel *profileTextFindFriends;
@property (nonatomic, strong) UILabel *profileTextSyncContact;
/// UIImageView: Vector
@property (nonatomic, strong) UIImageView *profileVector;

@end

@implementation ProfileGroup3

- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = [UIColor colorWithRed:1.00 green:1.00 blue:1.00 alpha:1.00];
        self.layer.cornerRadius = 10;
        self.layer.shadowOpacity = 1;
        self.layer.shadowRadius = 4;
        self.layer.shadowOffset = CGSizeMake(0, 2);
        self.layer.shadowColor = [UIColor colorWithRed:0.00 green:0.00 blue:0.00 alpha:0.25].CGColor;
        [self setupSubviews];
        [self setupConstraints];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
}


- (void)setupSubviews {
    [self addSubview:self.profileVector];
    [self addSubview:self.profileImageFindFriend];
    [self addSubview:self.profileTextFindFriends];
    [self addSubview:self.profileTextSyncContact];
}

- (void)setupConstraints {
    [self.profileImageFindFriend mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.mas_left).offset(14);
        make.width.mas_equalTo(18);
        make.top.equalTo(self.mas_top).offset(17);
        make.height.mas_equalTo(18);
    }];

    [self.profileTextFindFriends mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.profileImageFindFriend.mas_right).offset(20);
        make.centerY.equalTo(self.profileImageFindFriend.mas_top).offset(3);
    }];

    [self.profileTextSyncContact mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.profileTextFindFriends.mas_left);
        make.centerY.equalTo(self.profileImageFindFriend.mas_bottom).offset(-0.5);
    }];

    [self.profileVector mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.profileTextFindFriends.mas_right).offset(9);
        make.width.mas_equalTo(10.17);
        make.top.equalTo(self.profileTextFindFriends.mas_centerY).offset(1);
        make.height.mas_equalTo(10.17);
    }];
}




- (UIImageView *)profileImageFindFriend {
    if (!_profileImageFindFriend) {
        _profileImageFindFriend = [[UIImageView alloc] initWithFrame:CGRectZero];
        _profileImageFindFriend.image = [UIImage imageNamed:@"profileImageFindFriend_2401@2x.png"];
        _profileImageFindFriend.contentMode = UIViewContentModeScaleAspectFit;
    }
     return _profileImageFindFriend;
}

- (UILabel *)profileTextFindFriends {
    if (!_profileTextFindFriends) {
        _profileTextFindFriends = [[UILabel alloc] initWithFrame:CGRectZero];
        _profileTextFindFriends.text = @"Find friends on Snapchat";
        _profileTextFindFriends.textAlignment = NSTextAlignmentLeft;
        _profileTextFindFriends.font = [UIFont fontWithName:@"Calibri" size:15] ?: [UIFont systemFontOfSize:15];
        _profileTextFindFriends.numberOfLines = 1;
        _profileTextFindFriends.textColor = [UIColor colorWithRed:0.00 green:0.00 blue:0.00 alpha:1.00];
    }
    return _profileTextFindFriends;
}

- (UILabel *)profileTextSyncContact {
    if (!_profileTextSyncContact) {
        _profileTextSyncContact = [[UILabel alloc] initWithFrame:CGRectZero];
        _profileTextSyncContact.text = @"Tap to sync your contact";
        _profileTextSyncContact.textAlignment = NSTextAlignmentLeft;
        _profileTextSyncContact.font = [UIFont fontWithName:@"Calibri-Bold" size:12] ?: [UIFont systemFontOfSize:12];
        _profileTextSyncContact.numberOfLines = 1;
        _profileTextSyncContact.textColor = [UIColor colorWithRed:0.77 green:0.77 blue:0.77 alpha:1.00];
    }
    return _profileTextSyncContact;
}

- (UIImageView *)profileVector {
    if (!_profileVector) {
        _profileVector = [[UIImageView alloc] initWithFrame:CGRectZero];
        _profileVector.image = [UIImage imageNamed:@"profileVector_2399@2x.png"];
        _profileVector.contentMode = UIViewContentModeScaleAspectFit;
    }
     return _profileVector;
}

@end

File: snapchat/profile/swift/CodiaDemoSwift/src/profileProfile/ProfileGroup3.swift
//
//  ProfileGroup3.swift
//
//  Created by codia-figma
//

import UIKit
import SnapKit


class ProfileGroup3: UIView {

    /// UIImageView: find friend
    lazy var profileImageFindFriend: UIImageView = {
        let view = UIImageView()
        view.image = UIImage(named: "profileImageFindFriend_2846@2x.png")
        view.contentMode = .scaleAspectFill

        return view
    }()

    lazy var profileTextFindFriends: UILabel = {
        let label = UILabel()
        label.text = "Find friends on Snapchat"
        label.font = UIFont(name: "Calibri", size: 15) ?? UIFont.systemFont(ofSize: 15)
        label.textAlignment = .left
        label.textColor = UIColor(red: 0.00, green: 0.00, blue: 0.00, alpha: 1.00)
        label.numberOfLines = 1;
        return label
    }()

    lazy var profileTextSyncContact: UILabel = {
        let label = UILabel()
        label.text = "Tap to sync your contact"
        label.font = UIFont(name: "Calibri-Bold", size: 12) ?? UIFont.systemFont(ofSize: 12)
        label.textAlignment = .left
        label.textColor = UIColor(red: 0.77, green: 0.77, blue: 0.77, alpha: 1.00)
        label.numberOfLines = 1;
        return label
    }()

    /// UIImageView: Vector
    lazy var profileVector: UIImageView = {
        let view = UIImageView()
        view.image = UIImage(named: "profileVector_2844@2x.png")
        view.contentMode = .scaleAspectFill

        return view
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor(red: 1.00, green: 1.00, blue: 1.00, alpha: 1.00)
        self.layer.cornerRadius = 10
        self.layer.shadowOpacity = 1
        self.layer.shadowRadius = 4
        self.layer.shadowOffset = CGSize(width: 0, height: 2)
        self.layer.shadowColor = UIColor(red: 0.00, green: 0.00, blue: 0.00, alpha: 0.25).cgColor
        setupSubviews()
        setupConstraints()
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func layoutSubview() {
      super.layoutSubviews()
    }

    func setupSubviews() {
        self.addSubview(self.profileVector)
        self.addSubview(self.profileImageFindFriend)
        self.addSubview(self.profileTextFindFriends)
        self.addSubview(self.profileTextSyncContact)
    }

    func setupConstraints() {
        profileImageFindFriend.snp.makeConstraints { make in
            make.left.equalTo(self.snp.left).offset(14)
            make.width.equalTo(18)
            make.top.equalTo(self.snp.top).offset(17)
            make.height.equalTo(18)
        }

        profileTextFindFriends.snp.makeConstraints { make in
            make.left.equalTo(profileImageFindFriend.snp.right).offset(20)
            make.centerY.equalTo(profileImageFindFriend.snp.top).offset(3)
        }

        profileTextSyncContact.snp.makeConstraints { make in
            make.left.equalTo(profileTextFindFriends.snp.left)
            make.centerY.equalTo(profileImageFindFriend.snp.bottom).offset(-0.5)
        }

        profileVector.snp.makeConstraints { make in
            make.left.equalTo(profileTextFindFriends.snp.right).offset(9)
            make.width.equalTo(10.17)
            make.top.equalTo(profileTextFindFriends.snp.centerY).offset(1)
            make.height.equalTo(10.17)
        }
    }
}